subsys/bluetooth/shell/bt.c

```diff
@@ -787,6 +787,101 @@ static int cmd_select(const struct shell *shell, size_t argc, char *argv[])
 	return 0;
 }
 
+static const char *get_conn_type_str(u8_t type)
+{
+	switch (type) {
+	case BT_CONN_TYPE_LE: return "LE";
+	case BT_CONN_TYPE_BR: return "BR/EDR";
+	case BT_CONN_TYPE_SCO: return "SCO";
+	default: return "Invalid";
+	}
+}
+
+static const char *get_conn_role_str(u8_t role)
+{
+	switch (role) {
+	case BT_CONN_ROLE_MASTER: return "master";
+	case BT_CONN_ROLE_SLAVE: return "slave";
+	default: return "Invalid";
+	}
+}
+
+static int cmd_info(const struct shell *shell, size_t argc, char *argv[])
+{
+	struct bt_conn *conn = NULL;
+	char addr_str[BT_ADDR_LE_STR_LEN];
+	bt_addr_le_t addr;
+	struct bt_conn_info info;
+	int err;
+
+	switch (argc) {
+	case 1:
+		if (default_conn) {
```

Comment:
Perhaps by default we should list all connections not just the default.

Reply:
How about adding "all" as argument to print all connections?

Reply:
I would need #17426 to iterate over all connections.